okapi/services/fieldnotes/upload/docs.xml

```diff
@@ -0,0 +1,49 @@
+<xml>
+    <brief>Upload Fieldnotes</brief>
+    <issue-id>630</issue-id>
+    <desc>
+        <p>Upload a set of one or more fieldnotes records.</p>
+    </desc>
+    <req name='field_notes'>
+        <p>Fieldnotes consist of one or more records in CSV format. Each
+        record comprises a geocache log consisting of four fields:</p>
+        <ul>
+            <li>Geocache Code</li>
+            <li>Date</li>
+            <li>Log Type</li>
+            <li>Log Text</li>
+        </ul>
+        <p>The first three fields are simple entities, the Log Text field is
+        different and a bit difficult as it may spread over muliple lines
+        and it may contain quote characters. In order to preserve the structure
+        of the records, the <i>field_notes</i> parameter must be passed as a
+        <b>base64 encoded utf8 string</b>. UTF-16LE, UTF-16BE with or without
+        BOM are not supported.
+        </p>
+        <p>Since the log type is passed as a string, its value must match the
+        values supported by the platform (case sensitive!):
+        <pre>
+[
+    "Found it",
+    "Didn't find it",
+    "Comment",
+    "Attended",
+    "Will attend",
+    "Archived",
+    "Ready to search",
+    "Temporarily unavailable"
+]
+        </pre>
+        </p>
+        <p>Note: This service method is not supported on all installations</p>
+    </req>
+    <common-format-params/>
+    <returns>
+        <p>A dictionary of the following structure:</p>
+        <ul>
+            <li>success - true</li>
+            <li>totalRecords - number of records in <i>field_notes</i></li>
+            <li>processedRecords - number of records inserted into the database</li>
+        </ul>
```

Comment:
This implies that some records might not be inserted. That makes sense, because it's always possible for the caller to give us invalid entries which cannot be inserted. However, we should clarify when and why this may happen. Also, success should probably not be true in this case?

Reply:
There will be records in that dataset that we need to skip. Cachers using those tools, e.g. cgeo, do caching on geocaching.com as well as opencaching.de and perhaps even more platforms all at the same time. The resulting fieldnotes dataset will have records that start with "GC....,....,..." and "OC.....,....,....". The PR filters out everything that is irrelevant for OCDE and will inform the user in the result via those counters.

Even if the record starts with "OC" it may still be discarded if the Log Type is not known to OCDE. Unfortunately there is no standardization across platforms and giving an ASCII text as log type is a really bad choice but it is the way it is and we can't do anything about it except discarding what doesn't fit into the platform.

Reply:
Thanks! Can you clarify all that in the file?

Reply:
done